Sources/StreamChat/Controllers/PollController/PollController.swift

```diff
@@ -0,0 +1,291 @@
+
+                delegate.pollController(self, didUpdateCurrentUserVotes: changes)
+            }
+        }
+
+        return observer
+    }()
+    
+    var _basePublishers: Any?
+    /// An internal backing object for all publicly available Combine publishers. We use it to simplify the way we expose
+    /// publishers. Instead of creating custom `Publisher` types, we use `CurrentValueSubject` and `PassthroughSubject` internally,
+    /// and expose the published values by mapping them to a read-only `AnyPublisher` type.
+    @available(iOS 13, *)
+    var basePublishers: BasePublishers {
+        if let value = _basePublishers as? BasePublishers {
+            return value
+        }
+        _basePublishers = BasePublishers(controller: self)
+        return _basePublishers as? BasePublishers ?? .init(controller: self)
+    }
+    
+    private let environment: Environment
+        
+    init(client: ChatClient, messageId: MessageId, pollId: String, environment: Environment = .init()) {
+        self.client = client
+        self.messageId = messageId
+        self.pollId = pollId
+        self.environment = environment
+        ownVotesQuery = PollVoteListQuery(
+            pollId: pollId,
+            optionId: nil,
+            pagination: .init(pageSize: 25, cursor: nil),
+            filter: .and(
+                [.equal(.userId, to: client.currentUserId ?? ""), .equal(.pollId, to: pollId)]
+            )
+        )
+        pollsRepository = client.pollsRepository
+        
+        super.init()
+    }
+    
+    override public func synchronize(_ completion: ((_ error: Error?) -> Void)? = nil) {
+        startObserversIfNeeded()
+
+        pollsRepository.queryPollVotes(query: ownVotesQuery) { [weak self] result in
+            guard let self else { return }
+            if let error = result.error {
+                self.state = .remoteDataFetchFailed(ClientError(with: error))
+            } else {
+                self.state = .remoteDataFetched
+            }
+            self.callback { completion?(result.error) }
+        }
+    }
+    
+    /// Casts a vote for a poll.
+    ///
+    /// - Parameters:
+    ///   - answerText: An optional text answer for the poll.
+    ///   - optionId: An optional identifier for the poll option.
+    ///   - completion: A closure to be called upon completion, with an optional `Error` if something went wrong.
+    public func castPollVote(
+        answerText: String?,
+        optionId: String?,
+        completion: ((Error?) -> Void)? = nil
+    ) {
+        if answerText == nil && optionId == nil {
+            completion?(ClientError.InvalidInput())
+            return
+        }
+        
+        pollsRepository.castPollVote(
+            messageId: messageId,
+            pollId: pollId,
+            answerText: answerText,
+            optionId: optionId,
+            currentUserId: client.currentUserId,
+            query: ownVotesQuery,
+            completion: completion
```

Comment:
We should use `callback {}` here for completion handler.

Reply:
fixed